components/kyma-environment-broker/internal/storage/driver/memory/instance.go

```diff
@@ -233,12 +233,9 @@ func (s *instances) filterInstances(filter dbmodel.InstanceFilter) []internal.In
 	equal := func(a, b string) bool {
 		return a == b
 	}
-	domainMatch := func(url, filter string) bool {
-		// Preceeding character is either a . or / (after protocol://)
-		// match subdomain inputs
-		// match any .upperdomain zero or more times
-		matchExpr := fmt.Sprintf(`[./]%s(\.[0-9A-Za-z-]+)*$`, filter)
-		matched, err := regexp.MatchString(matchExpr, url)
+	shootMatch := func(shootName, filter string) bool {
+		matchExpr := fmt.Sprintf(`^(%s)$`, filter)
+		matched, err := regexp.MatchString(matchExpr, shootName)
```

Comment:
Isn't it a regex for doing == on two strings?

Reply:
it is, wanted to keep it consistent with logic working on db:
https://github.com/kyma-project/control-plane/pull/1667/files#diff-5471acbb37034ddb3ff40a442be71d095a25404327018bb15c7afb79f6fc5961R721
I will change it to single string comparison